NMS-16142: Update UI README for dev workflow, pinia, etc. (#6685)
OpenNMS/opennms: ui/README.md

# Vue 3 + Typescript + Vite

> This template should help get you started developing with Vue 3, Typescript 4.9 and Vite 4.

Reviewer: ```suggestion
This template should help you start developing with Vue 3, Typescript 4.9, and Vite 4.
```

yarn build
```

Build/watch in dev mode
```
yarn watch:dev
```

Run unit tests
```
yarn test
```

## State management: Vuex and pinia
*NOTE:* Vuex is being replaced by [pinia](https://pinia.vuejs.org/).

This project uses [Vuex](https://next.vuex.vuejs.org/) with the modules pattern.
Each store module has separate files for state, actions, and mutations.

> Current convention is to only call actions from components, (no mutations).

Reviewer: ```suggestion
Current convention is to call actions only from components, (no mutations).
```

Vuex stores are under `store`; pinia stores are under `stores`.

## Vue-router

> Project routes make use of [vue-router](https://next.router.vuejs.org/guide/)

Reviewer: ```suggestion
Project routes make use of [vue-router](https://next.router.vuejs.org/guide/).
```

## Recommended IDE Setup

[VSCode](https://code.visualstudio.com/)

## Developer workflow

> Developer workflow for development -> build -> fast deploy. There may be issues running the vite development server since it runs on a different port than OpenNMS and OpenNMS menu items and URLs will be pointing to that port instead of the vite port.

Reviewer: ```suggestion
Developer workflow for development -> build -> fast deploy. There may be issues running the vite development server since it runs on a different port than OpenNMS; OpenNMS menu items and URLs will point to that port instead of the vite port.
```

- assuming your local OpenNMS Horizon code is in `~/projects/opennms`

- write your code

- from your `~/projects/opennms/ui` directory, run `yarn dev` or `yarn watch:dev` to build in development mode (or `yarn build` for production / minified mode)

> - have a console window open in the target directory where the built/deployed files need to be, e.g. `~/projects/opennms/target/opennms-33.0.0-SNAPSHOT/jetty-webapps/opennms/ui`

Reviewer: ```suggestion
- have a console window open in the target directory where the built/deployed files need to be, e.g., `~/projects/opennms/target/opennms-33.0.0-SNAPSHOT/jetty-webapps/opennms/ui`
```

- after build completes, run `cp ~/projects/opennms/ui/dist/assets/*.* assets` to copy all the built JS and other asset files, then run `cp ~/projects/opennms/ui/dist/index.html .` to copy the new `index.html` file

- from `target/snapshot/jetty-webapps/opennms/ui`, occasionally run `rm assets/*.*` to clear out old files

> - refresh your browser, which is pointed at `http://localhost:8980/opennms/ui/index.html#/nodes` or similar

Reviewer: ```suggestion
- refresh your browser, which points at `http://localhost:8980/opennms/ui/index.html#/nodes` or similar
```
